### PR TITLE
update karpenter-core

### PR DESCRIPTION
**Pull request by bwagner5**
Fixes # 

**Description**
 - Release v0.27.4 which adds some additional logging and metrics

**How was this change tested?**

* `make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bwagner5 — COMMENTED** [review]
/karpenter snapshot

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/PuerkitoBio/goquery v1.8.1
 	github.com/avast/retry-go v3.0.0+incompatible
 	github.com/aws/aws-sdk-go v1.44.195
-	github.com/aws/karpenter-core v0.27.3
+	github.com/aws/karpenter-core v0.27.4-0.20230516213910-fd3f7d15c696
```
Comment (jonathan-innis) [inline go.mod:10]: Should these pin the version `v0.27.4` directly?

Comment (bwagner5) [reply go.mod:10]: I wanted to run the e2e tests first... but seems snapshot doesn't work when not destined for main :(

Comment (jonathan-innis) [reply go.mod:10]: Wait, really? That's super weird :(

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Comment by jonathan-innis** [comment]
@bwagner5 Looks like it ran when I just called it. Not sure what was happening before, but I looked at the action and there isn't anything in there that should guard against it running the snapshot on other PRs for other branches

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Comment by jonathan-innis** [comment]
Ah, it's failing because it can't reference the `install-deps` action on the main branch. Do you think we could take the #2268 with this changes on the release branch so we have a way to build snapshots here?

**Review by bwagner5 — COMMENTED** [review]
/karpenter snapshot

**Comment by bwagner5** [comment]
:white_check_mark: machine-pr-3900-lmxsw tests passed





[14:55](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353309196889)
:white_check_mark: utilization-pr-3900-bvqg5 tests passed
[14:56](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353395467809)
:white_check_mark: ipv6-pr-3900-blbmq tests passed
[14:57](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353454394319)
:white_check_mark: drift-pr-3900-vxr5t tests passed
[14:59](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353592489029)
:white_check_mark: interruption-pr-3900-m8tcp tests passed
[15:02](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353742872529)
:white_check_mark: chaos-pr-3900-4zvbq tests passed
[15:03](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684353805641319)
:white_check_mark: consolidation-pr-3900-7q22s tests passed
[15:49](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684356556717109)
:white_check_mark: upgrade-pr-3900-fktqf tests passed
[15:50](https://amzn-aws.slack.com/archives/C03M9SFQHUM/p1684356602109369)
:white_check_mark: integration-pr-3900-glb9h tests passed

**Review by njtran — APPROVED** [review]

**Pull request merged**